src/lib.rs

```diff
@@ -1122,7 +1163,44 @@ impl<T: Clone> Receiver<T> {
     /// assert_eq!(r2.recv().await, Err(RecvError::Closed));
     /// # });
     /// ```
-    pub fn recv(&mut self) -> Recv<'_, T> {
+    pub fn recv(&mut self) -> Pin<Box<Recv<'_, T>>> {
+        Box::pin(self.recv_direct())
+    }
+
+    /// Receives a message from the channel without pinning the future to the heap.
+    ///
+    /// If the channel is empty, this method waits until there is a message.
+    ///
+    /// If the channel is closed, this method receives a message or returns an error if there are
+    /// no more messages.
+    ///
+    /// If this receiver has missed a message (only possible if overflow mode is enabled), then
+    /// this method returns an error and readjusts its cursor to point to the first available
+    /// message.
+    ///
+    /// The future returned by this method is not `Unpin` and must be pinned before use. This is
+    /// the desired behavior if you just `.await` on the future. For other uses cases, use the
+    /// [`recv`] method instead.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # futures_lite::future::block_on(async {
+    /// use async_broadcast::{broadcast, RecvError};
+    ///
+    /// let (s, mut r1) = broadcast(1);
+    /// let mut r2 = r1.clone();
+    ///
+    /// assert_eq!(s.broadcast(1).await, Ok(None));
+    /// drop(s);
+    ///
+    /// assert_eq!(r1.recv_direct().await, Ok(1));
+    /// assert_eq!(r1.recv_direct().await, Err(RecvError::Closed));
+    /// assert_eq!(r2.recv_direct().await, Ok(1));
+    /// assert_eq!(r2.recv_direct().await, Err(RecvError::Closed));
+    /// # });
+    /// ```
```

Comment:
nit: I'd rather not repeat the whole documentation, since we can just link to `recv` method that's right in the same document. I know that's how its been done for many methods but perhaps I shouldn't have followed that convention set in the existing code.